src/pudl/output/pudltabl.py

```diff
@@ -169,255 +170,237 @@ def pu_ferc1(self, update=False):
             )
         return self._dfs["pu_ferc1"]
 
-    ###########################################################################
-    # EIA 861 Interim Outputs (awaiting full DB integration)
-    ###########################################################################
-    def etl_eia861(
-        self, eia861_settings: Eia861Settings = Eia861Settings(), update: bool = False
-    ):
-        """A single function that runs the temporary EIA 861 ETL and sets all DFs.
-
-        This is an interim solution that provides a (somewhat) standard way of accessing
-        the EIA 861 data prior to its being fully integrated into the PUDL database. If
-        any of the dataframes is attempted to be accessed, all of them are set. Only
-        the tables that have actual transform functions are included, and as new
-        transform functions are completed, they would need to be added to the list
-        below. Surely there is a way to do this automatically / magically but that's
-        beyond my knowledge right now.
-
-        Args:
-            eia861_settings: Object containing validated settings relevant to EIA 861.
-            update: Whether to overwrite the existing dataframes if they exist.
-        """
-        if isinstance(self.ds, Datastore):
-            pass
-        elif self.ds is None:
-            pudl_settings = pudl.workspace.setup.get_defaults()
-            if pudl_settings["pudl_in"] is None:
-                raise FileNotFoundError(
-                    "In order to run the ad-hoc EIA-861 ETL PUDL needs a valid "
-                    "Datastore, but none was found. Run 'pudl_setup --help' "
-                    "to see how to create one."
-                )
-            self.ds = Datastore(local_cache_path=pudl_settings["data_dir"])
-        else:
-            raise TypeError(
-                "PudlTabl needs a PUDL Datastore object, but we got "
-                f"a {type(self.ds)}."
-            )
-
-        if update or self._dfs["balancing_authority_eia861"] is None:
-            logger.warning("Running the interim EIA 861 ETL process!")
-
-            eia861_raw_dfs = pudl.extract.eia861.Extractor(self.ds).extract(
-                settings=eia861_settings
-            )
-            self._dfs.update(
-                pudl.transform.eia861.transform(eia861_raw_dfs, eia861_settings)
-            )
-
-    def advanced_metering_infrastructure_eia861(
-        self, eia861_settings: Eia861Settings = Eia861Settings(), update=False
-    ):
+    def advanced_metering_infrastructure_eia861(self) -> pd.DataFrame:
         """An interim EIA 861 output function."""
-        self.etl_eia861(update=update)
-        return self._dfs["advanced_metering_infrastructure_eia861"]
+        return (
+            pd.read_sql("advanced_metering_infrastructure_eia861", self.pudl_engine)
+            .convert_dtypes(convert_floating=False)
+            .pipe(apply_pudl_dtypes, group="eia")
+        )
```

Comment:
Having 29 identical 5-line methods seems super dumb. Is there some way we can have a "method factory" inside the class that automatically defines all these methods just based on a list of table names?

Reply:
I thought about doing this, but I think it's basically `apply_pudl_dtypes()` I added `convert_dtypes()` too because I'm paranoid and it's probably not required.

Reply:
Sure! I'm not entirely sure either so I'd have to play around with some options. We could collaborate in a separate PR.

Reply:
Definitely separate! But would be fun to learn. So much duplication here for so little information. Seems silly.